filesystem/squashfs/squashfs.go

```diff
@@ -532,7 +532,7 @@ func (fs *FileSystem) hydrateDirectoryEntries(entries []*directoryEntryRaw) ([]*
 		body, header := in.getBody(), in.getHeader()
 		xattrIndex, has := body.xattrIndex()
 		xattrs := map[string]string{}
-		if has && xattrIndex != noXattrInodeFlag {
+		if has {
```

Comment:
When i was tracing down the bug i noticed all `xattrIndex` already do this check or return false so i removed it